pkg/common/catalog/plugin.go

```diff
@@ -144,7 +144,7 @@ func warnIfDeprecated(log logrus.FieldLogger, thisVersion, latestVersion Version
 	if thisVersion.Deprecated() {
 		log = log.WithField(telemetry.DeprecatedServiceName, thisVersion.New().GRPCServiceName())
 		if !latestVersion.Deprecated() {
-			log.WithField(telemetry.PreferredServiceName, latestVersion.New().GRPCServiceName())
+			log = log.WithField(telemetry.PreferredServiceName, latestVersion.New().GRPCServiceName())
```

Comment:
This bug was preventing the preferred service name from being logged appropriatly.